nemo/collections/tts/models/ed_mel2spec.py

```diff
@@ -0,0 +1,335 @@
+
+        for (k, s) in self.f_specs:
+            new_loss = self.calc_loss_smooth(x_spec, y_spec, T_ys, k, s)
+            output[f'loss_{k}_{s}'] = new_loss
+            loss = loss + new_loss
+
+        output['val_loss'] = loss
+
+        return output
+
+    def validation_epoch_end(self, outputs):
+        tensorboard_logs = {}
+
+        for k in outputs[0].keys():
+            tensorboard_logs[k] = torch.stack([x[k] for x in outputs]).mean()
+
+        return {'val_loss': tensorboard_logs['val_loss'], 'log': tensorboard_logs}
+
+    def __setup_dataloader_from_config(self, cfg, shuffle_should_be: bool = True, name: str = "train"):
+        if "dataset" not in cfg or not isinstance(cfg.dataset, DictConfig):
+            raise ValueError(f"No dataset for {name}")  # TODO
+        if "dataloader_params" not in cfg or not isinstance(cfg.dataloader_params, DictConfig):
+            raise ValueError(f"No dataloder_params for {name}")  # TODO
+        if shuffle_should_be:
+            if 'shuffle' not in cfg.dataloader_params:
+                logging.warning(
+                    f"Shuffle should be set to True for {self}'s {name} dataloader but was not found in its "
+                    "config. Manually setting to True"
+                )
+                with open_dict(cfg["dataloader_params"]):
+                    cfg.dataloader_params.shuffle = True
+            elif not cfg.dataloader_params.shuffle:
+                logging.error(f"The {name} dataloader for {self} has shuffle set to False!!!")
+        elif not shuffle_should_be and cfg.dataloader_params.shuffle:
+            logging.error(f"The {name} dataloader for {self} has shuffle set to True!!!")
+
+        dataset = instantiate(cfg.dataset)
+        return torch.utils.data.DataLoader(dataset, collate_fn=dataset.collate_fn, **cfg.dataloader_params)
+
+    def setup_training_data(self, cfg):
+        self._train_dl = self.__setup_dataloader_from_config(cfg)
+
+    def setup_validation_data(self, cfg):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg, shuffle_should_be=False, name="validation")
+
+    @classmethod
+    def list_available_models(cls) -> 'List[PretrainedModelInfo]':
+        """
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
+        Returns:
+            List of available pre-trained models.
+        """
+        list_of_models = []
+
+        # model = PretrainedModelInfo(
+        #     pretrained_model_name="DeepGriffinLim-fft_1024-22050Hz",
+        #     location="https://nemo-public.s3.us-east-2.amazonaws.com/nemo-1.0.0alpha-tests/DeepGriffinLim-fft_1024.nemo",  ##FIXME
+        #     description="The model is trained on LJSpeech sampled at 22050Hz, n_fft=1024, with 6 layers and a widening factor of 16. Can be used as a trained alternative for GriffinLim",
+        # )
+        # list_of_models.append(model)
+        return list_of_models
```

Comment:
Do you have a pretrained model for this?

Reply:
It is being trained